src/lib/util/resolveSwagger.ts

```diff
@@ -0,0 +1,131 @@
+import * as jsonParser from "@ts-common/json-parser"
+import * as jsonPointer from "json-pointer"
+import * as json from "@ts-common/json"
+
+import { readFileSync, writeFileSync } from 'fs'
+import { cloneDeep, Data,FilePosition,getFilePosition} from "@ts-common/source-map"
+import * as sm from "@ts-common/string-map"
+import  { toArray } from "@ts-common/iterator"
+import { pathToJsonPointer } from './utils'
+
+
+/*
+ * Merges source object into the target object
+ * @param {object} source The object that needs to be merged
+ *
+ * @param {object} target The object to be merged into
+ *
+ * @returns {object} target - Returns the merged target object.
+ */
+export function mergeObjects<T extends sm.MutableStringMap<Data>>(source: T, target: T): T {
+  const result: sm.MutableStringMap<Data> = target
+  for (const [key, sourceProperty] of sm.entries(source)) {
+    if (Array.isArray(sourceProperty)) {
+      const targetProperty = target[key]
+      if (!targetProperty) {
+        result[key] = sourceProperty
+      } else if (!Array.isArray(targetProperty)) {
+        throw new Error(
+          `Cannot merge ${key} from source object into target object because the same property ` +
+            `in target object is not (of the same type) an Array.`
+        )
+      } else {
+        result[key] = mergeArrays(sourceProperty, targetProperty)
+      }
+    } else {
+      result[key] = cloneDeep(sourceProperty)
+    }
+  }
+  return result as T
+}
+
+/*
+ * Merges source array into the target array
+ * @param {array} source The array that needs to be merged
+ *
+ * @param {array} target The array to be merged into
+ *
+ * @returns {array} target - Returns the merged target array.
+ */
+export function mergeArrays<T extends Data>(source: ReadonlyArray<T>, target: T[]): T[] {
+  if (!Array.isArray(target) || !Array.isArray(source)) {
+    return target
+  }
+  source.forEach(item => {
+    target.push(cloneDeep(item))
+  })
+  return target
+}
+
+export class ResolveSwagger {
+    innerSwagger:json.Json | undefined
+    file:string
+
+    constructor(file:string) {
+        this.file = file.replace('\\','/')
```

Comment:
Why replace here? Node standard path library supports some cross-platform function.

Reply:
just  make our output path have the same format

Reply:
I mean you can use path.resolve to make file format in different platform. Windox and Linux have different paths format .

Reply:
great . fix it